pluton/spawn/containercache/sshcache.go

```diff
@@ -0,0 +1,278 @@
+package containercache
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/coreos/mantle/platform"
+	"github.com/coreos/mantle/util"
+)
+
+const (
+	cacheDir          = "/home/core/containercache"
+	downloadRetries   = 3
+	downloadRetryWait = 15 * time.Second
+	downloadTimeout   = 2 * time.Minute
+)
+
+var once sync.Once
+
+type ImageName struct {
+	Name   string
+	Engine string // docker or rkt
+}
+
+// StartBastionOnce will populate a bastion machine with the given containers
+// for rkt and docker. Since its best to run this only once per test suite, the
+// bastion machine should be in a separate platform.Cluster then the one used
+// in the test so it can be available for all tests in a given run. This
+// function is guarded with a sync.Once, so only the first within a test suite
+// it will have it run.
+func StartBastionOnce(bastion platform.Machine, names []ImageName) error {
```

Comment:
Once advantage of calling `StartBastionOnce` during the k8s cluster setup code is that it gives us the opportunity to intelligently figure out what containers to prefetch. Right now most things are hardcoded besides bootkube and hyperkube. 

Soon these tests will begin using a bootkube binary instead of a container which means we have the opportunity to move the calling of `StartBastionOnce` after we render the bootkube assets. The bootkube assets contain all the names of containers we need and it wouldn't be difficult to parse them out. This moves us away from a static, hard-coded list of containers to prefetch.